src/Models/LoginSession.php

```diff
@@ -305,4 +306,33 @@ public static function getSessionLifetime(): int
 
         return LoginSession::config()->get('default_session_lifetime');
     }
+
+    /**
+     * Update LastAccessed date and IP address
+     *
+     * @param HTTPRequest $request
+     * @return void
+     */
+    public function updateLastAccessed(HTTPRequest $request = null)
+    {
+        $this->LastAccessed = DBDatetime::now()->Rfc2822();
+        $this->IPAddress = $this->getIpFromRequest($request);
+        $this->write();
+    }
+
+    /**
+     * @param HTTPRequest|null $request
+     * @return string
+     */
```

Comment:
```suggestion
```
This information is already in the typehints - this duplication is likely to get outdated and doesn't provide any additional information